Chris Frey <cdfrey@foursquare.net>

This patch checks for the presense of a forward slash at the beginning of
each path.  If found, it forces the prefix to "file://".

This is helpful when running the evolution2 plugin with the KitchenSync
GUI (snapshot 2007/07/12), since the GUI seems to strip off the file://
prefix no matter what you type into the fields, and the plugin needs
the prefix to operate properly.

An example of a path:
	file:///home/cdfrey/.evolution/addressbook/local/system


diff -ru libopensync-plugin-evolution2-0.22/src/evolution2_xml.c libopensync-plugin-evolution2-0.22-cdf/src/evolution2_xml.c
--- libopensync-plugin-evolution2-0.22/src/evolution2_xml.c	2007-03-27 07:49:55.000000000 -0400
+++ libopensync-plugin-evolution2-0.22-cdf/src/evolution2_xml.c	2007-07-27 22:47:43.000000000 -0400
@@ -59,13 +59,13 @@
 		char *str = (char*)xmlNodeGetContent(cur);
 		if (str) {
 			if (!xmlStrcmp(cur->name, (const xmlChar *)"address_path")) {
-				env->addressbook_path = g_strdup(str);
+				env->addressbook_path = g_strdup_printf("%s%s", (str[0] == '/') ? "file://" : "", str);
 			}
 			if (!xmlStrcmp(cur->name, (const xmlChar *)"calendar_path")) {
-				env->calendar_path = g_strdup(str);
+				env->calendar_path = g_strdup_printf("%s%s", (str[0] == '/') ? "file://" : "", str);
 			}
 			if (!xmlStrcmp(cur->name, (const xmlChar *)"tasks_path")) {
-				env->tasks_path = g_strdup(str);	
+				env->tasks_path = g_strdup_printf("%s%s", (str[0] == '/') ? "file://" : "", str);
 			}
 			xmlFree(str);
 		}
